scripts/rebalance-queue-masters

```diff
@@ -100,18 +100,19 @@ set_temp_queue_policies()
 
         # List of masters extracted from node_list + list_queues masters.
         IFS=$'\n'
-        for tmp in $(rabbitmqctl -sqp "$vhost" list_queues pid | sed -e 's/[><]//g' -e 's/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g' | uniq -c )
+        for tmp in $(echo "${!node_list[*]}" | uniq -c) $(rabbitmqctl -q -p "$vhost" list_queues pid | sed -e 's/[><]//g' -e 's/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g' | uniq -c )
```

Comment:
I don't see why this is necessary. Please give the output of `rabbitmqctl -q -p "$vhost" list_queues pid` showing why or a series of steps to demonstrate why.

I tested using a two-node cluster where, initially, all queue masters were on one node.

Reply:
I've tested with a 3 nodes cluster:
```
root@rabbitmq01 ~ # rabbitmqctl -q -p "/" list_queues pid | sed -e 's/[><]//g' -e 's/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g' | sort | uniq -c
   3937 rabbit@rabbitmq01
    385 rabbit@rabbitmq02
   1065 rabbit@rabbitmq03
root@rabbitmq01 ~ # ssh rabbitmq02 reboot
Connection to rabbitmq02 closed by remote host.
root@rabbitmq01 ~ # rabbitmqctl -q -p "/" list_queues pid | sed -e 's/[><]//g' -e 's/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g' | sort | uniq -c
   3937 rabbit@rabbitmq01
   1066 rabbit@rabbitmq03
```

Reply:
Right, but this shouldn't end up in miscalculation, because of the code to which I linked [here](https://github.com/rabbitmq/support-tools/blob/gh-13-review/scripts/rebalance-queue-masters#L87).

* Do you see an error when you run this version of the script? Does re-balancing not work?
* If you'd like, please run the script from the `gh-13-review` like this in your environment and attach the output to a follow-up comment:

```
/bin/bash -x ./rebalance-queue-masters > /tmp/rebalance.txt 2>&1
```

Reply:
Output of `/bin/bash -x ./rebalance-queue-masters > /tmp/rebalance.txt 2>&1`

```
root@rabbitmq01 ~ # /bin/bash -x ./rebalance-queue-masters.4
+ set -o errexit
+ set -o nounset
+ [[ 4 -lt 4 ]]
+ declare -r debug=false
+ declare -r 'orig_IFS= 	
'
+ declare -r default_vhost=/
+ declare -r 'default_queue_regex=.*'
+ declare -r default_disable_health_check=false
+ declare -ri default_max_count=60
+ declare -a nodes
+ declare -i node_count=0
+ main
+ parse_options
+ local OPTARG
+ local OPTIND
+ local opt
+ local vhost_opt_set=false
+ local queue_regex_opt_set=false
+ local disable_health_check_opt_set=false
+ local max_count_opt_set=false
+ getopts dhc:p:r: opt
+ shift 0
+ [[ false == \f\a\l\s\e ]]
+ declare -gr vhost=/
+ [[ false == \f\a\l\s\e ]]
+ declare -gr 'queue_regex=.*'
+ [[ false == \f\a\l\s\e ]]
+ declare -gr disable_health_check=false
+ [[ false == \f\a\l\s\e ]]
+ declare -gri max_count=60
+ validate
+ for cmd in erl rabbitmqctl
+ command -v erl
+ for cmd in erl rabbitmqctl
+ command -v rabbitmqctl
+ local -i i=0
++ rabbitmqctl eval 'rabbit_nodes:all_running().'
++ sed -e 's/[]['\'']//g' -e 's/,/ /g'
+ for node in '$(rabbitmqctl eval '\''rabbit_nodes:all_running().'\'' | sed -e "s/[]['\'']//g" -e '\''s/,/ /g'\'')'
+ nodes[$i]=rabbit@rabbitmq02
+ ((  i = i + 1  ))
+ for node in '$(rabbitmqctl eval '\''rabbit_nodes:all_running().'\'' | sed -e "s/[]['\'']//g" -e '\''s/,/ /g'\'')'
+ nodes[$i]=rabbit@rabbitmq03
+ ((  i = i + 1  ))
+ for node in '$(rabbitmqctl eval '\''rabbit_nodes:all_running().'\'' | sed -e "s/[]['\'']//g" -e '\''s/,/ /g'\'')'
+ nodes[$i]=rabbit@rabbitmq01
+ ((  i = i + 1  ))
+ node_count=3
+ ((  node_count < 2  ))
+ set_temp_queue_policies
++ now
++ date +%Y%m%d-%H:%M:%S
+ echo '20190305-07:37:08 [INFO] Setting temporary policies on vhost: /'
20190305-07:37:08 [INFO] Setting temporary policies on vhost: /
+ ensure_node_health
++ mktemp
+ local tmp=/tmp/tmp.cbYznvcj2T
+ for n in '"${nodes[@]}"'
+ rabbitmqctl -n rabbit@rabbitmq02 node_health_check
+ for n in '"${nodes[@]}"'
+ rabbitmqctl -n rabbit@rabbitmq03 node_health_check
+ for n in '"${nodes[@]}"'
+ rabbitmqctl -n rabbit@rabbitmq01 node_health_check
+ rm -f /tmp/tmp.cbYznvcj2T
+ true
+ local node=
+ local tmp=
+ local -i num_nodes=0
+ unset node_list
+ unset node_master_count
+ local -A node_list
+ local -A node_master_count
+ IFS=,
++ rabbitmqctl eval 'rabbit_nodes:all_running().'
++ sed -e 's/[]['\'']//g'
+ for tmp in '$(rabbitmqctl eval '\''rabbit_nodes:all_running().'\'' | sed -e "s/[]['\'']//g")'
+ node_list[$tmp]=1
+ node_master_count[$tmp]=0
+ ((  num_nodes = num_nodes + 1  ))
+ for tmp in '$(rabbitmqctl eval '\''rabbit_nodes:all_running().'\'' | sed -e "s/[]['\'']//g")'
+ node_list[$tmp]=1
+ node_master_count[$tmp]=0
+ ((  num_nodes = num_nodes + 1  ))
+ for tmp in '$(rabbitmqctl eval '\''rabbit_nodes:all_running().'\'' | sed -e "s/[]['\'']//g")'
+ node_list[$tmp]=1
+ node_master_count[$tmp]=0
+ ((  num_nodes = num_nodes + 1  ))
+ IFS=' 	
'
+ [[ false == \t\r\u\e ]]
+ IFS='
'
++ rabbitmqctl -sqp / list_queues pid
++ sed -e 's/[><]//g' -e 's/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g'
++ uniq -c
Error: could not recognise command
+ for tmp in '$(rabbitmqctl -sqp "$vhost" list_queues pid | sed -e '\''s/[><]//g'\'' -e '\''s/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g'\'' | uniq -c )'
+ IFS=' 	
'
+ set -- 1 Usage:
+ [[ false == \t\r\u\e ]]
+ node_master_count[$2]=1
+ IFS='
'
+ for tmp in '$(rabbitmqctl -sqp "$vhost" list_queues pid | sed -e '\''s/[><]//g'\'' -e '\''s/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g'\'' | uniq -c )'
+ IFS=' 	
'
+ set -- 1 rabbitmqctl '[-n' 'node]' '[-t' 'timeout]' '[-q]' command '[command' 'options]'
+ [[ false == \t\r\u\e ]]
+ node_master_count[$2]=1
+ IFS='
'
+ for tmp in '$(rabbitmqctl -sqp "$vhost" list_queues pid | sed -e '\''s/[><]//g'\'' -e '\''s/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g'\'' | uniq -c )'
+ IFS=' 	
'
+ set -- 1
+ [[ false == \t\r\u\e ]]
+ node_master_count[$2]=1
./rebalance-queue-masters.4: line 114: $2: unbound variable
```

1) The error is because of `-sqp` option
2) If you don't `sort` the command, the `uniq -c` does worth 

Can you test my PR version?

Reply:
Please note that you can attach files to GitHub comments. There is no need to paste the output.

Let me revise this pull request to remove the `-s` option.

Reply:
Great!

(I've attached the output, sorry for pasting directly.)

Reply:
I've removed `-s` and added `sort` prior to the `uniq -c` call, thanks for noticing that. Please give the latest version on the `gh-13-review` branch a try.